acceptancetests/jujupy/client.py

```diff
@@ -1462,6 +1462,10 @@ def _get_substrate_constraints(self):
             # For now only maas support spaces in a meaningful way.
             return 'spaces={}'.format(','.join(
                 '^' + space for space in sorted(self.excluded_spaces)))
+        elif self.env.lxd:
+            # LXD should be constrained by memory when running in HA, otherwise
+            # mongo just eats everything.
+            return 'mem=6G'
```

Comment:
@hmlanigan we need to set an upper limit.

Reply:
+1